src/handlers/channels/STAGE_INSTANCE_CREATE.ts

```diff
@@ -0,0 +1,26 @@
+import { eventHandlers } from "../../bot.ts";
+import { cacheHandlers } from "../../cache.ts";
+import type { StageInstance } from "../../types/channels/stage_instance.ts";
+import type { DiscordGatewayPayload } from "../../types/gateway/gateway_payload.ts";
+import { snowflakeToBigint } from "../../util/bigint.ts";
+
+export async function handleStageInstanceCreate(data: DiscordGatewayPayload) {
+  const { channelId, guildId, ...rest } = data.d as StageInstance;
```

Comment:
Please avoid unnecessary variable declaration, just use `payload.guildId` etc.

Reply:
this does not seem to be resolved

Reply:
> just use `payload.guildId` etc.
@itohatweb, there is no such variable `payload`.

Reply:
change this `const { channelId, guildId, ...rest } = data.d` to `const payload = data.d`

Reply:
> change this `const { channelId, guildId, ...rest } = data.d` to `const payload = data.d`

In that case, there will be an extra `guildId` and `channelId` field in the event handler argument/parameter.

Reply:
Its not really important if there is or not imo. At the end the types already say it different. Or even edit the type cause why removing it. Just we decided to not do this if the variable is only going to be used once or so.

Reply:
> Its not really important if there is or not imo. At the end the types already say it different. Or even edit the type cause why removing it. Just we decided to not do this if the variable is only going to be used once or so.

In that case, the properties `guildId` and `channelId` become nugatory since `guild.id` and `channel.id` always exist and channel are already theoretically present. On a very large bot, this can make a significant difference in terms of memory and storage.

Reply:
We do this on every other event too. Doing the same for this event won't hurt much.